FastSimulation/SimplifiedGeometryPropagator/plugins/FastSimProducer.cc

```diff
@@ -0,0 +1,573 @@
+    			if(!myFSimTrack.onVFcal())
+				{
+					myFSimTrack.setVFcal(PP, 0);
+				}
+    		}
+
+	    	// not necessary to continue propagation
+	    	if(caloLayer->getCaloType() == fastsim::SimplifiedGeometry::VFCAL)
+			{
+				caloLayer = nullptr;
+				break;
+			}
+
+	    	continue;
+	    }
+
+	    // Stupid variable used by the old propagator
+	    // For details check BaseParticlePropagator.h
+	    int success = 0;
+	    if(caloLayer->isForward())
+	    {
+	    	success = 2;
+	    	// particle moves inwards
+	    	if(particle->position().Z() * particle->momentum().Z() < 0)
+	    	{
+	    		success *= -1;
+	    	}
+	    }
+	    else
+	    {
+	    	success = 1;
+	    	// particle moves inwards
+	    	if(particle->momentum().X() * particle->position().X() + particle->momentum().Y() * particle->position().Y() < 0)
+	    	{
+	    		success *= -1;
+	    	}
+	    }
+
+	    // Save the hit
+	    if(caloLayer->getCaloType() == fastsim::SimplifiedGeometry::PRESHOWER1)
+		{	
+			if(myFSimTrack.onLayer1())
+			{
+				//caloLayer = 0;
+				//break;
+			}
+			else
+			{
+				myFSimTrack.setLayer1(PP, success);
+			}
+		}
+
+		if(caloLayer->getCaloType() == fastsim::SimplifiedGeometry::PRESHOWER2)
+		{					
+			if(myFSimTrack.onLayer2())
+			{
+				//caloLayer = 0;
+				//break;
+			}
+			else
+			{
+				myFSimTrack.setLayer2(PP, success);
+			}
+		}
+
+	    if(caloLayer->getCaloType() == fastsim::SimplifiedGeometry::ECAL)
+		{					
+			if(myFSimTrack.onEcal())
+			{
+				//caloLayer = 0;
+				//break;
+			}
+			else
+			{
+				myFSimTrack.setEcal(PP, success);
+			}
+		}
+
+	    if(caloLayer->getCaloType() == fastsim::SimplifiedGeometry::HCAL)
+		{					
+			if(myFSimTrack.onHcal())
+			{
+				//caloLayer = 0;
+				//break;
+			}
+			else
+			{
+				myFSimTrack.setHcal(PP, success);
+			}
+		}
+
+		if(caloLayer->getCaloType() == fastsim::SimplifiedGeometry::VFCAL)
+		{					
+			if(myFSimTrack.onVFcal())
+			{
+				//caloLayer = 0;
+				//break;
+			}
+			else
+			{
+				myFSimTrack.setVFcal(PP, success);
+			}
+		}
+
+		// Particle reached end of detector
+		if(caloLayer->getCaloType() == fastsim::SimplifiedGeometry::VFCAL)
+		{
+			caloLayer = nullptr;
+			break;
+		}
+	    
+	    LogDebug(MESSAGECATEGORY) << "--------------------------------"
+				      << "\n-------------------------------";
+	}
+
+	// do decays
+	// don't have to worry about daughters if particle already within the calorimetry
+	// since they will be rejected by the vertex cut of the ParticleFilter
+	if(!particle->isStable() && particle->remainingProperLifeTimeC() < 1E-10)
+	{
+	    LogDebug(MESSAGECATEGORY) << "Decaying particle...";
+	    std::vector<std::unique_ptr<fastsim::Particle> > secondaries;
+	    decayer_.decay(*particle,secondaries, _randomEngine->theEngine());
+	    LogDebug(MESSAGECATEGORY) << "   decay has " << secondaries.size() << " products";
+	    particleManager->addSecondaries(particle->position(), particle->simTrackIndex(),secondaries);
+	}
```

Comment:
This is necessary since after the particle has been propagated through the tracker (related to L293), we also have to further propagate the particle through the calorimetry in order to interface the new framework with the CalorimetryManager. I wanted to keep both propagations seperate from another since this makes it a lot easier to port the CalorimeterManager to the new Framework at a later stage.